java-checks/src/main/java/org/sonar/java/filters/SuppressWarningFilter.java

```diff
@@ -0,0 +1,155 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.java.filters;
+
+import com.google.common.collect.DiscreteDomains;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Multimap;
+import com.google.common.collect.Ranges;
+import com.google.common.collect.Sets;
+
+import org.sonar.api.issue.Issue;
+import org.sonar.api.rule.RuleKey;
+import org.sonar.api.utils.AnnotationUtils;
+import org.sonar.check.Rule;
+import org.sonar.java.checks.SuppressWarningsCheck;
+import org.sonar.java.model.JavaTree;
+import org.sonar.java.model.LiteralUtils;
+import org.sonar.java.syntaxtoken.LastSyntaxTokenFinder;
+import org.sonar.plugins.java.api.JavaCheck;
+import org.sonar.plugins.java.api.tree.AnnotationTree;
+import org.sonar.plugins.java.api.tree.ClassTree;
+import org.sonar.plugins.java.api.tree.ExpressionTree;
+import org.sonar.plugins.java.api.tree.LiteralTree;
+import org.sonar.plugins.java.api.tree.MethodTree;
+import org.sonar.plugins.java.api.tree.NewArrayTree;
+import org.sonar.plugins.java.api.tree.Tree;
+import org.sonar.plugins.java.api.tree.VariableTree;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Set;
+
+public class SuppressWarningFilter extends BaseTreeVisitorIssueFilter {
+
+  private static final String SUPPRESS_WARNING_RULE_KEY = getSuppressWarningRuleKey();
+
+  private static String getSuppressWarningRuleKey() {
+    return AnnotationUtils.getAnnotation(SuppressWarningsCheck.class, Rule.class).key();
+  }
+
+  @Override
+  public Set<Class<? extends JavaCheck>> filteredRules() {
+    return ImmutableSet.of();
+  }
+
+  @Override
+  public boolean accept(Issue issue) {
+    if (issueShouldNotBeReported(issue, excludedLinesByRule())) {
+      return false;
+    }
+    return true;
+  }
+
+  private static boolean issueShouldNotBeReported(Issue issue, Multimap<String, Integer> excludedLineByRule) {
+    RuleKey issueRuleKey = issue.ruleKey();
+    for (String excludedRule : excludedLineByRule.keySet()) {
+      if (("all".equals(excludedRule) || isRuleKEy(excludedRule, issueRuleKey)) && !isSuppressWarningRule(issueRuleKey)) {
+        Collection<Integer> excludedLines = excludedLineByRule.get(excludedRule);
+        if (excludedLines.contains(issue.line())) {
+          return true;
+        }
+      }
+    }
+    return false;
+  }
+
+  private static boolean isRuleKEy(String rule, RuleKey ruleKey) {
+    try {
+      // format of the rules requires a repository: "repo:key"
+      return ruleKey.equals(RuleKey.parse(rule));
+    } catch (IllegalArgumentException e) {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1166)